django/core/cache/backends/base.py

```diff
@@ -265,6 +265,16 @@ def decr_version(self, key, delta=1, version=None):
         """
         return self.incr_version(key, -delta, version)
 
+    def touch(self, key, timeout=DEFAULT_TIMEOUT, version=None):
+        """Touch the supplied key and set its expiry time to timeout. Returns True
+        if the key was successfully touched, False if the key did not exist.
+        """
+        if self.has_key(key, version):
+            self.set(key, self.get(key, version=version), timeout=timeout, version=version)
```

Comment:
This implementation has a race condition if another thread does `set` / `delete` on the same key in between the set and get here. I would feel more comfortable with it just raising `NotImplementedError` and allowing the backends to pass through to their proper `touch` command.